src/main/java/com/fasterxml/jackson/databind/deser/BasicDeserializerFactory.java

```diff
@@ -482,7 +482,9 @@ protected void _addImplicitConstructorCreators(DeserializationContext ctxt,
         final AnnotationIntrospector intr = ccState.annotationIntrospector();
         final VisibilityChecker<?> vchecker = ccState.vchecker;
         List<AnnotatedWithParams> implicitCtors = null;
-        final boolean preferPropsBased = config.getConstructorDetector().singleArgCreatorDefaultsToProperties();
+        final boolean preferPropsBased = config.getConstructorDetector().singleArgCreatorDefaultsToProperties()
+                // [databind#3968]: Only Record's canonical constructor is allowed to be considered for properties-based creator
```

Comment:
You mean, implicitly? And this because by default, canonical constructor is already considered (usually) Properties-based?

Reply:
Yes, canonical constructor will be recognised as properties-based creator by:
1. https://github.com/FasterXML/jackson-databind/blob/a1e6208fd8c6d1873ae41106eb58f8c3751a12f3/src/main/java/com/fasterxml/jackson/databind/deser/BasicDeserializerFactory.java#L1025-L1033
2. https://github.com/FasterXML/jackson-databind/blob/a1e6208fd8c6d1873ae41106eb58f8c3751a12f3/src/main/java/com/fasterxml/jackson/databind/deser/BasicDeserializerFactory.java#L542-L548

...because only a canonical constructor will have an implicit constructor + `propDef` with implicit name.

Reply:
Ok. I think I understand how this works around the issue, although it still seems like it shouldn't be needed (but is due to... combination of factors). But since it actually resolves an actual problem & does not break other tests, I think I should merge this. And revisit whole thing for 2.16; but not block 2.15.x.